test/runtests.jl

```diff
@@ -47,3 +47,10 @@ end
 @test _macroexpand( :(@pipe a|>b|>c(_) ) ) == :(c(b(a)))
 @test _macroexpand( :(@pipe a|>b(x,_)|>c|>d(_,y) ) ) == :(d(c(b(x,a)),y))
 @test _macroexpand( :(@pipe a|>b(xb,_)|>c|>d(_,xd)|>e(xe) |>f(xf,_,yf)|>_[i] ) ) == :(f(xf,(e(xe))(d(c(b(xb,a)),xd)),yf)[i]) #Very Complex
+
+# broadcasting
+fn(x) = x^2
+@test _macroexpand( :(@pipe fn |> _.(1:2) ) ) == :(fn.(1:2))
```

Comment:
what happens if we do:
```julia
_macroexpand( :(@pipe fn .|> _.(1:2) ) )
```

Reply:
Now it produces `:(var"._".(1:2))` and `@pipe fn .|> _.(1:2)` crashes.
I'm not entirely sure, what is desired behaviour, how the resulting code should look like.
In plain Julia `fn .|> x->x.(1:2) == fn |> x->x.(1:2)`, and since `fn` isa not iterable AFAIK, then `fn ,|>` produces same result as `fn |>`, so I'm not sure where I want to go with this.

Reply:
we want to match the normal behavour.

Here is an interesting set of cases:
```
julia> [1,2,2] |> x-> atan.([10,20,30], x)
3-element Array{Float64,1}:
 1.4711276743037347
 1.4711276743037347
 1.5042281630190728

julia> [1,2,2] .|> x-> atan.([10,20,30], x)
3-element Array{Array{Float64,1},1}:
 [1.4711276743037347, 1.5208379310729538, 1.5374753309166493]
 [1.373400766945016, 1.4711276743037347, 1.5042281630190728]
 [1.373400766945016, 1.4711276743037347, 1.5042281630190728]

julia> [1,2,2] .|> x-> atan([10,20,30], x)
ERROR: MethodError: no method matching atan(::Array{Int64,1}, ::Int64)
```

Reply:
I see. If I understand it correctly, the embedding of input array is not able to cover cases 2 and 3, because the `.` notation is not expressive enough, so I should explicitly call broadcast function in the macro, right?
Because I can write test for `@test _macroexpand( :(@pipe [1,2,2] |> atan.([10,20,30], _) ) ) == :(atan.([10,20,30], [1,2,2]))`
but I'm kinda lost in decising what shouls `@test _macroexpand( :(@pipe [1,2,2] .|> atan.([10,20,30], _) ) )` equal to.